### PR TITLE
Refresh API token regularly

### PR DESCRIPTION
**Pull request by tvirot**
Warning: untested code!
[bitmoji](https://render.bitstrips.com/v2/cpanel/9669972-117823830_6-s4-v1.png?transparent=1&width=246)

### CODE REVIEW

File: youpin-api.js [hunk 1]
```diff
@@ -32,6 +32,27 @@ class Api {
     });
   }
 
+  refreshToken() {
+    // TO-DO: Call /token/auth/refresh instead
+    const app = feathers()
+      .configure(hooks())
+      .configure(rest(uri).superagent(request))
```
Comment (tvirot) [inline youpin-api.js]: Oops, this should be `this.uri`

File: youpin-api.js [hunk 2]
```diff
@@ -32,6 +32,27 @@ class Api {
...
+      .configure(authentication());
+
+    new Promise(resolve => {
```
Comment (supasate) [inline youpin-api.js]:
```
    new Promise((resolve, reject) => {
```

File: youpin-api.js [hunk 3]
```diff
@@ -32,6 +32,27 @@ class Api {
...
+      app.authenticate({
+        type: 'local',
+        'email': this.username,
+        'password': this.password
+      }).then(result => {
+        this.token = app.get('token');
+        resolve(this);
+      }).catch(error => {
+        console.log(error);
```
Comment (supasate) [inline youpin-api.js:57]:
```
 reject(error);
```

**Comment by parnurzeal** [comment]
The thing is that we don't actually need to use `feathers-authentication` lib at all. (sorry for my bad example :P).

We can do a normal http request like youpin-web and I think this is a good example.
https://github.com/youpin-city/youpin-web/blob/11153fbc155a3f40b12d157dd8d47786e26ac38e/lib/route/default.js#L194

Let me know which approach you want to go for.

**Comment by tvirot** [comment]
I just used your bad example for now to fix your shameful act #6 . Ideally, we should have only one API wrapper module for both bot and web app anyway, so let's fix that later!

**Comment by supasate** [comment]
Bad code that is based on bad example to fix shameful act looks good to me :P. 

[Update]
Just a bit wonder about the place of code calling `setInterval()` to refresh the token. It looks like random placement.

**Pull request merged**